harness/determined/util.py

```diff
@@ -246,6 +248,37 @@ def jsonable(obj: Any) -> Any:
     return json.dumps(jsonable(obj), indent=indent, sort_keys=sort_keys)
 
 
+def _make_shutil_ignore(root_path: pathlib.Path) -> Callable:
+    ignore = list(common_constants.DEFAULT_DETIGNORE)
+    ignore_path = root_path / ".detignore"
+    if ignore_path.is_file():
+        with ignore_path.open("r") as detignore_file:
+            ignore.extend(detignore_file)
+
+    import pathspec
+
+    ignore_spec = pathspec.PathSpec.from_lines(pathspec.patterns.GitWildMatchPattern, ignore)
```

Comment:
all this code is copy/pasted from somewhere relatively far away, can you create a helper function, just to keep it from diverging?  This is the kind of dusty old code where nobody will know that it was implemented in two different places 6 months down the road.

I would even like to see some sort of DetIgnore object or something like that, where the os.walk filter and the shutil ignore filter are implemented right next to each other.  Not because I feel like OO-design is really called for here, just to keep the two implementations as close to each other as possible.

Reply:
I've refactored out `detignore.py` with these filters. I also had to pull out `v1file_utils` with v1File helpers because it's now used both in `context.py` and `detignore.py`.